vaadin-text-field-flow/src/test/java/com/vaadin/flow/component/textfield/tests/NumberFieldTest.java

```diff
@@ -97,4 +97,21 @@ public void assertMaxValidation() {
         field.setValue(99.999);
         Assert.assertFalse(field.isInvalid());
     }
+
+    @Test
+    public void testPropertyValue() {
+        testValueProperty(1.0d, "1");
+        testValueProperty(2.0d, "2");
+        testValueProperty(5.0d, "5");
+        testValueProperty(9.0d, "9");
+        testValueProperty(21.4d, "21.4");
+        testValueProperty(123456789.01d, "123456789.01");
+        testValueProperty(-1.050d, "-1.05");
```

Comment:
Would like to see a test with e.g. `0.5`, so that it doesn't convert to `.5`.

Reply:
Done